app/dojo.py

```diff
@@ -0,0 +1,141 @@
+from room import LivingSpace, Office
+from person import Fellow, Staff
+from random import choice
+
+class Dojo:
+	"""
+	A class that contains the methods to create a new room and to add new people to the 
+	Dojo. 
+
+	"""
+
+	def __init__(self):
+
+		self.living_spaces = {"Amugune": ["Jonathan Swift", "Tom Jones"]}
+		self.offices = {"Saitoti": { "Fellows":["Jonathan Swift", "Tom Jones"], "Staff":[]}, "Sahari": {"Staff":["William Jakoyo", "Franklin Barasa"], "Fellows":[]} }
+		self.fellows = ["Jonathan Swift", "Tom Jones"]
+		self.staff = ["William Jakoyo", "Franklin Barasa"]
+
+	def create_room(self, room_type, room_name_list):
+		"""Create a new room."""
+
+		self.room_type = room_type
+
+		if self.room_type not in ["Office", "Living-space"]:
+			print("The Dojo doesn't have the room type you've entred")
+
+		else:
+			
+			for room_name in room_name_list:
+				self.room_name = room_name
+
+				"""Check whether the room already exists"""
+				if self.room_name in self.offices or self.room_name in self.living_spaces:
+					print("There is already a room with the name " + self.room_name + ". Please try again")
+					raise ValueError
+				
+				elif self.room_type == "Office":
+					"""Create room if it doesn't exists"""
+			
+					new_office = Office(self.room_name)
+					self.offices[str(new_office)] = new_office.residents
+					print("An office called " + self.room_name + " has been successfully created!")
+
+				else:
+					
+					new_living_space = LivingSpace(self.room_name)
+					self.living_spaces[str(new_living_space)] = new_living_space.residents
+					print("A living space called " + self.room_name + " has been successfully created!")
+
+	def add_new_member(self, person_category, person_name, wants_accomodation="N"):
+		"""Add a new member to the Dojo"""
+
+		self.person_name = person_name
+		self.person_category = person_category
+		self.wants_accomodation = wants_accomodation
+
+		"""Check if the person already exists"""
+		if self.person_name in self.fellows or self.person_name in self.staff:
```

Comment:
This line doesn't check if the person exists because you've not supposed to save person details in a dictionary but rather create an instance of either `Staff` or `Fellow` and append to list of either of fellows or staff.
Eg. `staff = Staff(first_name, last_name)` if the person_category is staff.